src/hooks/useQueryLogs.ts

```diff
@@ -17,8 +17,16 @@ const { setLogData } = logsStoreReducers;
 const { parseQuery } = filterStoreReducers;
 
 const appendOffsetToQuery = (query: string, offset: number) => {
-	const hasOffset = query.toLowerCase().includes('offset');
-	return !hasOffset ? query.replace(/offset\s+\d+/i, `OFFSET ${offset}`) : `${query}`;
+	const offsetRegex = /offset\s+\d+/i;
+	const limitRegex = /limit\s+\d+/i;
+
+	if (offsetRegex.test(query)) {
```

Comment:
Also consider SQL queries with offset

Reply:
It should work for the SQL queries with existing OFFSET query.